### PR TITLE
IGAPP-365: Remove fastlane-plugin-sentry

### PR DESCRIPTION
**Pull request by maxammann**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -612,9 +621,10 @@ jobs:
                         name: '[FL] TestFlight Upload'
                         working_directory: native
             - run:
-                command: bundle exec fastlane ios sentry_upload version_name:${NEW_VERSION_NAME} version_code:${NEW_VERSION_CODE} build_config_name:<< parameters.build_config_name >> sourcemap_path:attached_workspace/<< parameters.build_config_name >>.ios.bundle.map
-                name: '[FL] Upload to Sentry'
-                working_directory: native
+                command: |
+                    eval $(yarn workspace --silent build-configs --silent manage to-bash << parameters.build_config_name >> ios)
+                    tools/sentry-release "$BUILD_CONFIG_BUNDLE_IDENTIFIER" "${NEW_VERSION_NAME}" ~/attached_workspace/sourcemaps/<< parameters.build_config_name >> "${NEW_VERSION_CODE}"
+                name: Sentry Upload
```
Comment (Taggotty) [inline .circleci/config.yml:627]: Why did you change the naming of the command?

File: native/fastlane/read_build_config.rb [hunk 2]
```diff
@@ -1,11 +1,6 @@
...
 
-def read_build_config(build_config_name, platform, project_root='.')
-  json = yarn(
-      command: "babel-node tools/build-config to-json #{build_config_name} #{platform}",
-      project_root: project_root,
-      flags: "--silent"
-  )
-
+def read_build_config(build_config_name, platform)
+  json = `yarn workspace --silent build-configs --silent manage to-json #{build_config_name} #{platform}`
```
Comment (klinzo) [inline native/fastlane/read_build_config.rb:4]: Crazy, how does it now it should be executed?

File: .circleci/src/jobs/build_android.yml [hunk 3]
```diff
@@ -31,19 +31,24 @@ steps:
       name: "[FL] << parameters.build_config_name >> build"
       command: ./gradlew clean && bundle exec fastlane build version_name:${NEW_VERSION_NAME} version_code:${NEW_VERSION_CODE} build_config_name:<< parameters.build_config_name >>
       working_directory: native/android
+  - run:
+      name: Prepare Sourcemaps
+      command: |
+        mkdir -p sourcemaps/<< parameters.build_config_name >>
+        cp native/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map sourcemaps/<< parameters.build_config_name >>/
```
Comment (Taggotty) [inline .circleci/src/jobs/build_android.yml:38]: before we used `move` for the sourcemaps. What was your resoning to use `copy` instead?

File: web/tools/webpack.config.babel.js [hunk 4]
```diff
@@ -76,14 +76,13 @@ const createConfig = (env: { config_name?: string, dev_server?: boolean, version
...
-    version = `${version}+${commitSha.substring(0, SHORT_COMMIT_SHA_LENGTH)}`
-  }
+  versionName = versionName ?? readVersionName()
+  commitSha = commitSha?.substring(0, SHORT_COMMIT_SHA_LENGTH) ?? 'Commit SHA unknown'
 
   console.log('Used config: ', buildConfigName)
   console.log('Configured as running in dev server: ', !devServer)
-  console.log('Version: ', version)
+  console.log('Version name: ', versionName)
+  console.log('Version nameCommit SHA: ', commitSha)
```
Comment (klinzo) [inline web/tools/webpack.config.babel.js:85]:
```suggestion
  console.log('Commit SHA ', commitSha)
```

File: web/tools/webpack.config.babel.js [hunk 5]
```diff
@@ -63,7 +63,7 @@ const generateManifest = (content: string, buildConfigName: string): string => {
 }
 
 const createConfig = (env: { config_name?: string, dev_server?: boolean, version_name?: string, commit_sha?: string } = {}) => {
-  const { config_name: buildConfigName, commit_sha: commitSha, version_name: versionName, dev_server: devServer } = env
+  let { config_name: buildConfigName, commit_sha: commitSha, version_name: versionName, dev_server: devServer } = env
```
Comment (klinzo) [inline web/tools/webpack.config.babel.js:66]: I'd like if this stays `const`, e.g.:
```suggestion
  let { config_name: buildConfigName, commit_sha: passedCommitSha, version_name: passedVersionName, dev_server: devServer } = env
```
And doing
```
  const versionName = passedVersionName ?? readVersionName()
  const commitSha = passedCommitSha?.substring(0, SHORT_COMMIT_SHA_LENGTH) ?? 'Commit SHA unknown'
```
below.

File: native/fastlane/Fastfile [hunk 6]
```diff
@@ -5,40 +5,6 @@ require_relative "read_build_config"
 require "yaml"
 require "base64"
 
-# The following parameters have to be passed:
```
Comment (klinzo) [inline native/fastlane/Fastfile:8]: :tada:

**Review by klinzo — APPROVED** [review]

File: native/fastlane/read_build_config.rb [hunk 7]
```diff
@@ -1,11 +1,6 @@
...
   JSON.parse(json)
```
Comment (Taggotty) [inline native/fastlane/read_build_config.rb:5]: What is that supposed to do?

**Review by Taggotty — APPROVED** [review]

File: native/fastlane/read_build_config.rb (hunk 2 above)
Comment (maxammann) [reply native/fastlane/read_build_config.rb:4]: backtick magic :D
thats how things work in ruby: magic

Comment (klinzo) [reply native/fastlane/read_build_config.rb:4]: Not sure if I like it :thinking:

File: native/fastlane/read_build_config.rb (hunk 7 above)
Comment (maxammann) [reply native/fastlane/read_build_config.rb:5]: returns the parsed result

File: .circleci/src/jobs/build_android.yml (hunk 3 above)
Comment (maxammann) [reply .circleci/src/jobs/build_android.yml:38]: not really a difference, but changing the build directory has the potential to go wrong.

File: .circleci/config.yml (hunk 1 above)
Comment (maxammann) [reply .circleci/config.yml:627]: Because it is no longer part of FastLane

File: native/fastlane/read_build_config.rb (hunk 2 above)
Comment (maxammann) [reply native/fastlane/read_build_config.rb:4]: github does not show the highlighting correct, looks different in intellij
